core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -2514,6 +2518,7 @@ private Mapping dynamicUpdate() {
     }
 
     public void testTranslogReplay() throws IOException {
+        final LongSupplier inSyncGlobalCheckpointSupplier = () -> this.engine.getLocalCheckpointTracker().getCheckpoint();
```

Comment:
I'm wondering if we should make this the default?

Reply:
Most of the time, we prefer to deal with an in-sync global checkpoint, thus making this as a default makes sense to me. I will do in a follow-up.